app/utils/hstasklisthelper/HsTaskListHelperUtils.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.hstasklisthelper
+
+
+import controllers.register.establishers.company.{routes => establisherCompanyRoutes}
+import models.register.Entity
+import models._
+import play.api.i18n.Messages
+import play.api.mvc.Call
+import utils.{Enumerable, UserAnswers}
+import identifiers.register.establishers.{company => establisherCompany}
+
+trait HsTaskListHelperUtils extends Enumerable.Implicits {
+
+  implicit val messages: Messages
+
+  def createSpoke(answers: UserAnswers,
+                  spokeName: String,
+                  mode: Mode, srn: Option[String], name: String, index: Int): EntitySpoke = {
+
+    val isChangeLink = getCompleteFlag(answers, index, spokeName)
+    val isComplete: Option[Boolean] = if (mode == NormalMode) isChangeLink else None
+
+    isChangeLink match {
+      case Some(true) => EntitySpoke(Link(getChangeLinkText(spokeName)(name), getChangeLink(spokeName)(mode, srn, index).url), isComplete)
+      case Some(false) => EntitySpoke(Link(getChangeLinkText(spokeName)(name), getAddLink(spokeName)(mode, srn, index).url), isComplete)
+      case None => EntitySpoke(Link(getAddLinkText(spokeName)(name), getAddLink(spokeName)(mode, srn, index).url), None)
+    }
+  }
+
+  def createDirectorPartnerSpoke(entityList: Seq[Entity[_]],
+                                 spokeName: String,
+                                 mode: Mode, srn: Option[String], name: String, index: Int): EntitySpoke = {
+
+    val isComplete: Option[Boolean] = if (mode == NormalMode && entityList.nonEmpty) Some(entityList.forall(_.isCompleted)) else None
+
+    if (entityList.isEmpty)
+      EntitySpoke(Link(getAddLinkText(spokeName)(name), getAddLink(spokeName)(mode, srn, index).url), None)
+    else
+      EntitySpoke(Link(getChangeLinkText(spokeName)(name), getChangeLink(spokeName)(mode, srn, index).url), isComplete)
+  }
+
+  private def getCompleteFlag(answers: UserAnswers, index: Int, spokeName: String): Option[Boolean] = spokeName match {
+    case "establisherCompanyDetails" => answers.get(establisherCompany.IsDetailsCompleteId(index))
```

Comment:
These spoke names seem to be repeated quite a few times: it might be a good idea to have a statlc set of spoke name constants.